src/dex/inception/inception.ts

```diff
@@ -0,0 +1,238 @@
+
+  async getPoolIdentifiers(
+    srcToken: Token,
+    destToken: Token,
+    side: SwapSide,
+    blockNumber: number,
+  ): Promise<string[]> {
+    return [
+      this.dexKey +
+        '-' +
+        [srcToken.address.toLowerCase(), destToken.address.toLowerCase()]
+          .sort((a, b) => (a > b ? 1 : -1))
+          .join('_'),
+    ];
+  }
+
+  async getPoolState(
+    pool: InceptionEventPool,
+    blockNumber: number,
+  ): Promise<PoolState> {
+    const eventState = pool.getState(blockNumber);
+    if (eventState) return eventState;
+    const onChainState = await pool.generateState(blockNumber);
+    pool.setState(onChainState, blockNumber);
+    return onChainState;
+  }
+
+  async getPricesVolume(
+    srcToken: Token,
+    destToken: Token,
+    amounts: bigint[],
+    side: SwapSide,
+    blockNumber: number,
+    limitPools?: string[],
+  ): Promise<null | ExchangePrices<InceptionDexData>> {
+    if (side === SwapSide.BUY) {
+      return null;
+    }
+
+    const src = getTokenFromAddress(this.network, srcToken.address);
+    const dest = getTokenFromAddress(this.network, destToken.address);
+
+    if (src !== dest) {
+      return null;
+    }
+
+    const poolState = await this.getPoolState(this.eventPool, blockNumber);
+    if (!poolState) return null;
+
+    return [
+      {
+        prices: amounts.map(amount => {
+          const ratio = poolState[src.symbol.toLowerCase()].ratio;
+          return (ratio * amount) / DECIMALS;
+        }),
+        unit: DECIMALS,
+        gasCost: 120_000,
+        exchange: this.dexKey,
+        data: {
+          exchange: dest.vault,
+        },
+        poolAddresses: [src.token.toLowerCase()],
+      },
+    ];
+  }
+
+  getCalldataGasCost(
+    poolPrices: PoolPrices<InceptionDexData>,
+  ): number | number[] {
+    return CALLDATA_GAS_COST.DEX_OVERHEAD + CALLDATA_GAS_COST.LENGTH_SMALL;
+  }
+
+  getAdapterParam(): AdapterExchangeParam {
+    return {
+      targetExchange: NULL_ADDRESS,
+      payload: '0x',
+      networkFee: '0',
+    };
+  }
+
+  getDexParam(
+    srcToken: Address,
+    destToken: Address,
+    srcAmount: NumberAsString,
+    destAmount: NumberAsString,
+    recipient: Address,
+    data: InceptionDexData,
+    side: SwapSide,
+  ): DexExchangeParam {
+    const dexParams = getTokenFromAddress(this.network, srcToken);
+
+    const isNative = dexParams.baseTokenSlug === 'ETH';
+    let swapData;
+    let dexFuncHasRecipient;
+    if (isNative) {
+      swapData = this.poolInterface.encodeFunctionData('stake()', []);
+      dexFuncHasRecipient = false;
+    } else {
+      swapData = this.vaultInterface.encodeFunctionData('deposit', [
+        srcAmount,
+        recipient,
+      ]);
+      dexFuncHasRecipient = true;
+    }
+
+    return {
+      needWrapNative: false,
+      dexFuncHasRecipient,
+      exchangeData: swapData,
+      targetExchange: dexParams.vault,
+      swappedAmountNotPresentInExchangeData: true,
+      returnAmountPos: undefined,
```

Comment:
Deposit function returns `iShares`, assume we can use it in the `returnAmountPos`

Reply:
Better to skip optimization for this case, since `iShares` might differ with real amount a user gets

<img width="879" alt="Screenshot 2024-09-10 at 14 44 02" src="https://github.com/user-attachments/assets/1ec80396-f829-4f95-bc35-fbdc8e4dad74">

Reply:
fixed